frontend/helpers/index.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * @param {Array} tags product tags
+ * @param {Array} triggerTags config trigger tags
+ * @returns {boolean}
+ */
+export const hasTriggerTag = (tags, triggerTags) => {
+  if (!tags) {
+    return false;
+  }
+
+  return tags.some(tag =>
+    triggerTags.some(trigger => trigger === tag));
+};
+
+/**
+ * @param {Array} additionalProperties product additional properties
+ * @param {Array} triggerProps config trigger props
+ * @returns {boolean}
+ */
+export const hasTriggerProp = (additionalProperties, triggerProps) => {
+  if (!additionalProperties) {
+    return false;
+  }
+
+  return additionalProperties.some(obj =>
+    triggerProps.some(trigger => trigger === obj.label));
```

Comment:
product properties have a label and a value. To tigger it should match both.  So the configuration for triggerProps will be an array of objects. The objects in the array will have a label and a value property. It is possible that a merchant would have a product property like showBadge that is assigned to all products. But the merchant will only want the badge shown for products that have the showBadge property with the value of 'yes.' This extension would so the badge on every product that had the showBadge property regardless of its value. This is not the way it should work.

Reply:
Gotcha. Updated